### DISCUSSION TITLE
Android Support

### DISCUSSION THREAD

**Discussion by jurises99** [SFTtech/openage#1393] [Ideas]
Will there be support on android in the future updates?,
Caesar III is on android using julius and it has a nice UI and controls, so maybe an android port of the openage?, It would be really nice to see this running on android platform.

**Comment by TobiasFP** [comment]
In short: no.
Explainer: I am not by any means a core part of the dev team, so i cannot make any final statements, but i do believe that i can give an answer that somebody from the core team can accept it elaborate on.
The game engine is written in c++ that could be fun under Android via the Android ndk. Some parts of the ui and a lot of helper stuff could also be run under the ndk, namely the parts that are converted to c coffee via the use of cython, however, a looooot of the code could not be run under Android (standard python parts in particular). This part is why i believe that the answer to your question is definitely a no, but even if we could magically fix that part, running most things under the Android ndk is not easily done, which just puts another nail in the coffin. A last (but truly unnecessary) nail in the coffin is the whole ui experience on touch interfaces.
This is just my two cents, i might be wrong but i doubt it.

But it was certainly a good question, so keep on asking :)

**Reply by roke-julian-lockhart** [reply]
@TobiasFP and @heinezen, would that not be true for [`issues/315`](https://github.com/SFTtech/openage/issues/315#issuecomment-112844749), too? I ask because the issue remains open, so I presume that it's feasible somehow.

**Reply by TheJJ** [reply]
if somebody thinks this is useful and does it, great. i wouldn't focus on this just now, but we've always kept in mind to not use weird technologies that prevent porting to other platforms if somebody tackles this.

**Comment by heinezen** [comment]
What @TobiasFP says is basically correct. The engine uses a mixture of C++ and Python. That would not really work well for Android.

Also, mobile is not the greatest platform for RTS, so we don't actively consider a port. But that's not saying it will never happen, it's just very unlikely. If we find a solution, then we'll ry to make it work :)
